js/src/admin/components/ExtensionPage.js

```diff
@@ -0,0 +1,251 @@
+import Alert from '../../common/components/Alert';
+import Page from '../../common/components/Page';
+import Button from '../../common/components/Button';
+import icon from '../../common/helpers/icon';
+import ItemList from '../../common/utils/ItemList';
+import listItems from '../../common/helpers/listItems';
+import LoadingModal from './LoadingModal';
+import ExtensionPermissionGrid from './ExtensionPermissionGrid';
+import addExtensionPermission from '../utils/addExtensionPermission';
+import Switch from '../../common/components/Switch';
+import saveSettings from '../utils/saveSettings';
+
+export default class ExtensionPage extends Page {
+  oninit(vnode) {
+    super.oninit(vnode);
+
+    this.settings = {};
+    this.loading = false;
+    this.extension = app.data.extensions[this.attrs.routeName];
```

Comment:
After thinking about this a bit more, I'm actually fine with something like this: regardless of how we get the extension ID, retrieving the extension object itself will work a bit like this.